engine/src/main/java/org/destinationsol/ui/nui/NUIManager.java

```diff
@@ -186,7 +197,10 @@ public void update(SolApplication solApplication) {
                 focusManager.getFocus().onKeyEvent(event);
             }
 
-            for (NUIScreenLayer uiScreen : uiScreens) {
+            // Create a copy of the list, as screens may be removed in response to key events.
+            // This does not happen for most key events, however it may occur with KeyActivatedButton instances,
+            // which already have special handling in the NUIScreenLayer class.
+            for (NUIScreenLayer uiScreen : new LinkedList<>(uiScreens)) {
```

Comment:
I'll assume there's a reason to create a new `LinkedList`...

Reply:
A new list is created because you can't change the contents of the list whilst iterating over it I thought that the comment above explained that but maybe it isn't clear. If there's a better way of doing this, then I would gladly change that code.

Reply:
I was more wondering why a `LinkedList` specifically. But looking at it now I see that I was just tired missed that `uiScreens` is already a `LinkedList`. 👍